optuna/integration/allennlp.py

```diff
@@ -111,6 +254,26 @@ def __init__(
         else:
             self._include_package = include_package
 
+        system_attrs = {
```

Comment:
As an alternative approach, we can get the storage URL without changing the storage API. How about separating the PR to implement `Storage.get_url()`?

```
storage = trial.study._storage
if isinstance(storage, RDBStorage):
    url = storage.url
elif isinstance(storage, RedisStorage):
    url = storage._url
elif isinstance(storage, _CachedStorage):
    assert isinstance(storage._backend, RDBStorage)
    url = storage._backend.url
else:
    raise ValueError("Storage must be RDBStorage or RedisStorage.")
```

Reply:
> How about separating the PR to implement Storage.get_url()?

Thank you for the suggestion, I agree with you.
I revised my PR (262f868 , 9b200af)  to remove `get_url` from each storage.

I rename `_url` in `RedisStorage` with `url` for consistency with `RDBStorage`. After this PR, I'll work on making `url` private in `RDBStorage` and `RedisStorage` at the same time.

Also, let me note `url` would be an empty string when no `url` can be loaded. It is because `AllenNLPExecutor` itself works on InMemoryStorage.

Reply:
Thanks a lot! Fewer files are now involved and it looks a bit cleaner. A small comment on the latest changes though, I think there's little benefit in aligning `RedisStorage`'s URL attribute to `RDBStorage` making it public. The `RDBStorage` has several of its attributes made public already and it's not just the `url`. We should probably limit the scope of the visibility matter to `RDBStorage`. It's always easier to make a private attribute public, than the other way around.

Reply:
Thank you for your suggestion. It makes sense. I made `url` private in `RedisStorage` at c537aa2.